sanic/request.py

```diff
@@ -53,6 +56,38 @@
     from json import loads as json_loads  # type: ignore
 
 
+@dataclasses.dataclass()
+class Credentials:
+    _auth_type: Optional[str]
+    token: Optional[str]
+    _username: Optional[str] = dataclasses.field(default=None)
+    _password: Optional[str] = dataclasses.field(default=None)
+
+    def __post_init__(self):
+        if self._auth_type == "Basic":
+            self._username, self._password = (
+                b64decode(self.token.encode("utf-8")).decode().split(":")
+            )
+
+    @property
+    def username(self):
+        if not self._check_auth_type:
+            raise AttributeError("Username is available for Basic Auth only")
+        return self._username
+
+    @property
+    def password(self):
+        if not self._check_auth_type:
+            raise AttributeError("Password is available for Basic Auth only")
+        return self._password
+
+    @property
+    def _check_auth_type(self) -> bool:
```

Comment:
A `check` attribute sounds more like a function than a property`. Also, the name does not really signify what it is actually doing.